File: 1.Data structure/6.PriorityQueue/이중 우선순위 큐.py
def solution(operations):
    answer = []
    for i in operations:
        temp = i.split(" ")
        if temp[0] == "I":
            answer.append(int(temp[1]))
        elif temp[0] == "D" and len(answer) > 0:
            answer.sort()
            if temp[1] == "-1":
                answer.pop(0)
            else:
                answer.pop()
    answer.sort()
    print(answer)
    return [answer[len(answer)-1] if len(answer)>0 else 0,answer[0] if len(answer)>0 else 0]

File: 1.Data structure/6.PriorityQueue/이중 우선순위 큐.java
import java.util.Collections;
import java.util.ArrayList;

class Solution {
    public int[] solution(String[] operations) {
        ArrayList<Integer> answer = new ArrayList<Integer>();
        for(String s : operations){
            String[] temp = s.split(" ");
            if (temp[0].equals("I")){
                answer.add(Integer.parseInt(temp[1]));
            }else if(temp[0].equals("D") && answer.size() > 0){
                Collections.sort(answer);
                if (temp[1].equals("-1")){
                    answer.remove(0);
                }else
                    answer.remove(answer.size()-1);
            }
        }
        if (answer.size() > 0){
            Collections.sort(answer);
            return new int[] {answer.get(answer.size()-1),answer.get(0)};
        }else{
            return new int[] {0,0};    
        }
    }
}